### ISSUE TITLE
A11y_Model Builder_Add Data_Input_Screen Reader: Narrator is not announcing the name of “column to predict” and “Time series column” combo box menu items.

### ISSUE THREAD

**Issue by manaswita-chougule** [dotnet/machinelearning-modelbuilder#2182]
### GitHubTags:
#A11yMAS; #A11yTCS; #A11ySev3; #DesktopApp; #Visual Studio Client; #SH-ModelBuilder-Win32-Jun2022; #Win32; #WCAG1.3.1; #Info And Relationship; #ED:Product:Narrator; #ED:Bug41132404; #External; #External:Narrator; #FirstParty:Bug41132404;

### Environment Details:
Application Name: Model Builder
Windows Version: Windows 11 22H2 OS Builder 25145.1011
Visual Studio: Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview Version 17.3.0 Preview 2.0

### Repro Steps:

1. Install Visual studio and open Visual studio
2. Navigate to C# project name and right click on it
3. Navigate to "Add" button and press enter
4. Navigate to "Machine learning model" and press enter
5. "Add new item" popup will open
6. Navigate to "Add" button and press enter
7. "Select scenario" popup will open and select "Forecasting" scenario
8. "Select training environment" page will open then select "Next step"
9. "Add Data" page will appear and navigate to "Input" section.
10. Tab till column to predict” and “Time series column” combo box and press enter.
11. Check whether Narrator is not announcing the name of “column to predict” and “Time series column” combo box menu items or not.

### Actual Result:
Narrator is not announcing the name of “column to predict” and “Time series column” combo box menu items.

### Expected Result:
Narrator should announce the name of “column to predict” and “Time series column” combo box menu items.

![#2182_narrator](https://user-images.githubusercontent.com/91586101/176687162-09084229-8f2c-4c4b-97d9-a90e84bd11ab.gif)

**Comment by beccamc** [comment]
Can you try this again with Version 17.3.0 Preview 3.0? It worked for me. If I went fast through the list then narrator didn't announce until something was selected. Going through the menu items at a normal speed read them out loud. 

![dropdown_working](https://user-images.githubusercontent.com/9122518/180863636-c7249b6e-95f2-4a17-85d6-ded80d64dcd4.gif)

**Comment by manaswita-chougule** [comment]
Hi @beccamc, verified the issue in below environment:
Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.3.0 Preview 4.0
OS : Windows 11 22H2 OS Build 25163.1000

The narrator is not announcing the combo box menu items, please find attachment for reference.
![2182_Not Fixed](https://user-images.githubusercontent.com/91586101/181210883-ee64cdf7-7053-48f8-9cc3-54231d8158f6.gif)

**Comment by beccamc** [comment]
@zewditu When you get back can you try to repro this? I can't get a repro and I'd like another set of eyes.

**Comment by zewditu** [comment]
@beccamc I am able to reproduce it .

**Comment by beccamc** [comment]
@zewditu Are you on Windows 11?

**Comment by zewditu** [comment]
Yes, I am  in Windows 11
________________________________

**Comment by beccamc** [comment]
@zewditu That might be the difference. Can you run Accessibility Insights for Windows on that page with Windows 11?

**Comment by beccamc** [comment]
@manaswita-chougule Can you try this in Windows 10? Can you also check your narrator verbosity settings too please? 

![image](https://user-images.githubusercontent.com/9122518/182694187-d07e42cc-abb1-4be2-bb73-2295ebc5557a.png)

**Comment by zewditu** [comment]
I run accessibility insight in windows 11, no issue found 
![image](https://user-images.githubusercontent.com/36615490/184226251-11e3a459-ad32-41c8-8709-10a7f37f0e08.png)

**Comment by rperez030** [comment]
@zewditu  The question is not whether issues were found. We need to check whether the right properties and patterns are being used so that we can tell if it is a narrator issue or an issue with the control. For example, does selection.Selection pattern updates as we select element in the ComboBox? Has this been tested with a different screen reader for comparison, for example NVDA?

**Comment by zewditu** [comment]
We discuss this offline with  @beccamc ,   when we make the Combo box property `IsEditable`  set to `True `  the narrator does not announce item names.

**Comment by rperez030** [comment]
Thanks @zewditu  for clarifying that. It pointed me in the right direction.

**Comment by beccamc** [comment]
This is a WPF issue. Here is the associated bug - https://github.com/dotnet/wpf/issues/7005

@manaswita-chougule Is this enough to close this bug? If not, are you able to test on Windows 10? It will work on Windows 10.

**Comment by anjalisheel-wpf** [comment]
@manaswita-chougule @beccamc I investigated this bug, it seems it is narrator's issue and not specific to wpf. I observed the same behavior for tooltip and textbox as well. Please open it for the narrator team.

Linking similar item for your reference  :  https://github.com/dotnet/wpf/issues/5557

**Comment by beccamc** [comment]
It looks like Narrator may already have this issue filed - https://microsoft.visualstudio.com/OS/_workitems/edit/41132404

**Comment by beccamc** [comment]
@manaswita-chougule Given the narrator bug above can we close this?

**Comment by manaswita-chougule** [comment]
As per the comment from Narrator team over the bug https://microsoft.visualstudio.com/OS/_workitems/edit/41132404, it seems the issue is not with narrator but with this application. Could you please look into this issue and provide us with the fixes.
Thanks!

**Comment by beccamc** [comment]
@manaswita-chougule I'm not sure which comment from Narrator team you are referring to. This is a problem with Narrator + Windows 11. We are able to reproduce this on a simple application. What is the appropriate way to file a Narrator bug so that we can close this issue? 

Running this program will reproduce the error when using Windows 11. [ComboBox.zip](https://github.com/dotnet/machinelearning-modelbuilder/files/9538156/ComboBox.zip)

**Comment by rperez030** [comment]
I'm sharing a readout of the events fired by the Combobox taken with the Inspect tool. The text portion of the Combobox is notifying about the changes in selection, and the ComboBox is firing the valueChange event containing the new value when I navigate using the arrow keys. @manaswita-chougule  unless the Narrator team can indicate the behavior that this control is failing, this should be resolved as a Narrator related bug. Narrator is not only not announcing the change in selection, it also cannot track the caret in the editable portion of the control.

[EventsLog.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/9538550/EventsLog.txt)

**Comment by beccamc** [comment]
Thank you @rperez030! Really appreciate it

**Comment by manaswita-chougule** [comment]
@beccamc As per Roberto’s comment, this is external to Narrator and not a product team bug. Since this is sev3, as per [Bug Severity Process](https://microsoft.sharepoint.com/teams/cea11y/SitePages/Bug-Severity-Process.aspx?xsdata=MDV8MDF8fGU5OTBkMWQ0NTk0NDRkZWQ3NWJlMDhkYTkxOGNlNDM1fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc5ODIzMzE1MzY1MDQyNTR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT18MXxNVGs2YldWbGRHbHVaMTlPZW14cFRtcENhazVxU1hSWlZGcG9XV2t3TUZwdFRYZE1WMFpwV1dwUmRGbFhXbWxOYW1ScFdtcENhazlIV20xQWRHaHlaV0ZrTG5ZeXx8&sdata=ZVlQeFZIcHFOVGNTQmVaa2c4NWxvWlpQbjl0eVgwSm4rSzVUMVlrNDVBYz0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cv-mchougule%40microsoft.com&OR=Teams-HL&CT=1662978674555&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiIyNy8yMjA4MTYwMTQwMiIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D) it won't affect to the compliance and product team can close it or wait for Narrator to fix. Narrator bugs do not count against compliance. Thanks!

**Comment by beccamc** [comment]
Great! If product team can close, we will close this bug. Thanks everyone!
